apps/mobile/src/services/notifications/backend.ts

```diff
@@ -79,34 +86,43 @@ export const registerForNotificationsOnBackEnd = async ({
   chainIds,
   fcmToken,
   notificationAccountType,
+  noAuth = false,
 }: {
   safeAddress: string
   signer: Wallet | HDNodeWallet | null
   chainIds: string[]
   fcmToken: string
   notificationAccountType: NOTIFICATION_ACCOUNT_TYPE
+  noAuth?: boolean
 }) => {
   const isOwner = notificationAccountType === NOTIFICATION_ACCOUNT_TYPE.OWNER
   const deviceUuid = await getDeviceUuid()
 
-  await authenticateSigner(signer, chainIds[0])
+  if (!noAuth) {
+    await authenticateSigner(signer, chainIds[0])
+  }
 
   const NOTIFICATIONS_GRANTED = isOwner ? OWNER_NOTIFICATIONS : REGULAR_NOTIFICATIONS
 
   await getStore()
     .dispatch(
-      notificationsApi.endpoints.notificationsUpsertSubscriptionsV2.initiate({
-        upsertSubscriptionsDto: {
-          cloudMessagingToken: fcmToken,
-          safes: chainIds.map((chainId) => ({
-            chainId,
-            address: safeAddress,
-            notificationTypes: NOTIFICATIONS_GRANTED,
-          })),
-          deviceType: isAndroid ? 'ANDROID' : 'IOS',
-          deviceUuid,
+      notificationsApi.endpoints.notificationsUpsertSubscriptionsV2.initiate(
+        {
+          upsertSubscriptionsDto: {
+            cloudMessagingToken: fcmToken,
+            safes: chainIds.map((chainId) => ({
+              chainId,
+              address: safeAddress,
+              notificationTypes: NOTIFICATIONS_GRANTED,
+            })),
+            deviceType: isAndroid ? 'ANDROID' : 'IOS',
+            deviceUuid,
+          },
         },
-      }),
+        {
+          forceOmitCredentials: noAuth,
```

Comment:
Not sure I understand this. If the user is not authenticated we call `authenticateSigner` further up but then also omit credentials. Why is this needed?

Reply:
Because I initially went for omit-ing the credentials, but this didn't work. I kept it, but also now log out the user before trying to access this endpoint.